flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/DefaultJobLeaderService.java

```diff
@@ -470,6 +475,7 @@ protected void onRegistrationFailure(Throwable failure) {
             return gateway.registerTaskManager(
                     taskManagerRpcAddress,
                     unresolvedTaskManagerLocation,
+                    jobId,
```

Comment:
Why is it not more appropriate to use the fencingToken for this?

Reply:
The `fencingToken` is a concept of the underlying `RpcSystem` whereas the `JobID` is a concept of the "application layer". That's why it is not guaranteed that fencing tokens are different iff `JobIDs` are different. An example is the standalone mode where the fencing token is always `0`.